src/Shipment.php

```diff
@@ -112,18 +114,37 @@ function (ResponseInterface $response) use ($request, $options) {
         };
     }
 
-    private function updateToken(RequestInterface $request)
+    private function getRefreshedToken(): string
     {
-        $token = $this->handleRefreshToken();
-        return \GuzzleHttp\Psr7\modify_request($request, [
-            'set_headers' => [
-                'Authorization' => 'Bearer ' . $token['access_token'],
-            ],
-        ]);
+        return $this->handleRefreshToken()['access_token'];
     }
 
     public function handleRefreshToken(): array
     {
         return $this->oAuth2->refreshToken($this->refreshToken);
     }
+
+    protected function createStack(): HandlerStack
```

Comment:
O que acha de trocar para esse nome ? Mais verboso, porém mais claro

```suggestion
    protected function createHttpClientMiddlewareStack(): HandlerStack
```

Reply:
Não posso trocar o nome desse método, porquê ele sobrescreve aquele `createStack` do `Shipment`. Ok, poderia até trocar lá também, mas aí cai [nesse comentário](https://github.com/melhorenvio/melhor-envio-sdk-php/pull/9#discussion_r1042458778)

Reply:
Acho que cai [nesse comentário](https://github.com/melhorenvio/melhor-envio-sdk-php/pull/9#discussion_r1042458778)